File: tests/test_cpg/fixtures/parity_open_redirect.py
"""Cross-language open redirect fixture — Python (Flask)."""

from flask import Flask, request, redirect

app = Flask(__name__)


@app.route("/login")
def login():
    next_url = request.args.get("next", "/")  # $ source=open_redirect
    return redirect(next_url)  # $ sink=open_redirect


@app.route("/goto")
def goto():
    target = request.args.get("url", "/")  # $ source=open_redirect
    return redirect(target, code=302)  # $ sink=open_redirect

File: tests/test_cpg/fixtures/parity_open_redirect.js
/** Cross-language open redirect fixture — JavaScript (Express). */

var req = { query: {}, body: {} };
var res = { redirect: function() {} };

function loginRedirect() {
    var next = req.query.next || '/';  // $ source=open_redirect
    res.redirect(next);  // $ sink=open_redirect
}

function gotoRedirect() {
    var target = req.query.url;  // $ source=open_redirect
    res.redirect(302, target);  // $ sink=open_redirect
}
